ut_core.py

```diff
@@ -0,0 +1,149 @@
+
+import socket
+import time
+import yaml
+import re
+
+class UTCoreMenuNavigator:
+    """
+    Navigates through the UTcore menu system, trigger the execution of test cases, and collect results.
+    """
+
+    def __init__(self, menu_config_path, console, test_profile_url=None):
+        """
+        Initializes the UTCoreMenuNavigator object with a menu configuration file and an optional test profile.
+
+        Args:
+            menu_config_path (str): The file path to the menu configuration YAML file.
+            console: The console session object to communicate with the UTcore system.
+            test_profile_url (str, optional): The URL to a test profile YAML file. Defaults to None.
+        """
+        self.session = console
+        self.menu_config = self.load_yaml(menu_config_path)
+        self.shell = self.session.open_interactive_shell()
+
+        if test_profile_url:
+            self.test_profile = self.load_yaml(test_profile_url)
+        else:
+            self.test_profile = None
+
+    
+    def load_yaml(self, path):
+        """
+        Loads and parses a YAML file from the specified path.
+
+        Args:
+            path (str): The file path to the YAML file.
+
+        Returns:
+            dict: The parsed YAML content.
+        """
+        with open(path, 'r') as file:
+            return yaml.safe_load(file)
+
+    
+    def navigate_to_test(self, group_name, test_name):
```

Comment:
This function looks like it still missing the logic to work out the number to use to get through the menu to teh desired test.